Resources/Computing/SingularityComputingElement.py

```diff
@@ -274,11 +285,29 @@ def submitJob(self, executableFile, proxy, jobDesc, log, logLevel, **kwargs):
     # Mount /cvmfs in if it exists on the host
     withCVMFS = os.path.isdir("/cvmfs")
     innerCmd = os.path.join(self.__innerdir, "dirac_container.sh")
-    cmd = ["singularity", "exec", "-C"]
+
+    singularityOpts = []
+    singularityOpts.append('exec')
+    singularityOpts.append('-c')
+    singularityOpts.append('-i')
+    singularityOpts.append('-p')
+    singularityOpts.append('-W "%s"' % baseDir)
     if withCVMFS:
```

Comment:
The reason to use "-C" instead of "-c -i -p". It is because in the latest version of singularity, the option "-C" has a slightly different, adding the function of cleaning up env before start singularity.  With "-C",  the DIRAC job can't get necessary env eg. X509_USER_PROXY